tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -72,178 +59,175 @@ setup() {
 
     # For debug sake
-    echo "Pod $kata_pod_without_nydus_config file:"
-    cat $kata_pod_without_nydus_config
-
-    add_allow_all_policy_to_yaml "$kata_pod_without_nydus_config"
-    k8s_create_pod "$kata_pod_without_nydus_config"
-
-    echo "Kata pod test-e2e without nydus annotation is running"
-    echo "Check the image was not pulled in the guest"
-    sandbox_id=$(get_node_kata_sandbox_id $node)
-    echo "sandbox_id is: $sandbox_id"
-
-    # The assert_rootfs_count will be FAIL.
-    # The expect count of rootfs in host is "2" but the found count of rootfs in host is "1"
-    # As the the first time we pull the $unencrypted_image_1 image via nydus-snapshotter in the guest
-    # for all subsequent pulls still use nydus-snapshotter in the guest
-    # More details: https://github.com/kata-containers/kata-containers/issues/8337
-    # The test case will be PASS after we use containerd 2.0 with 'image pull per runtime class' feature:
-    # https://github.com/containerd/containerd/issues/9377
-    assert_rootfs_count "$node" "$sandbox_id" "2"
-}
+    echo "Trusted storage $storage_config file:"
+    cat $storage_config
+    
+    # Create persistent volume and persistent volume claim
+    kubectl create -f $storage_config
+
+    pod_config=$(mktemp "${BATS_FILE_TMPDIR}/$(basename "${pod_config_template}").XXX")
+    IMAGE="$large_image" NODE_NAME="$node" envsubst < "$pod_config_template" > "$pod_config"
+
+    # Set a short CreateContainerRequest timeout in the annotation to fail to pull image in guest
+    create_container_timeout=10
+    set_metadata_annotation "$pod_config" \
+        "io.katacontainers.config.runtime.create_container_timeout" \
+        "${create_container_timeout}"
 
-@test "Test we can pull an other unencrypted image outside the guest and then inside the guest successfully" {
-    # 1. Create one kata pod with the $unencrypted_image_2 image and without nydus annotation
-    kata_pod_without_nydus_config="$(new_pod_config "$unencrypted_image_2" "kata-${KATA_HYPERVISOR}")"
-    set_node "$kata_pod_without_nydus_config" "$node"
-    set_container_command "$kata_pod_without_nydus_config" "0" "sleep" "30"
+    # Enable dm-integrity in guest
+    set_metadata_annotation "${pod_config}" \
+        "io.katacontainers.config.hypervisor.kernel_params" \
+        "agent.secure_storage_integrity=true"
+
+    # Set annotation to pull image in guest
+    set_metadata_annotation "${pod_config}" \
+        "io.containerd.cri.runtime-handler" \
+        "kata-${KATA_HYPERVISOR}"
 
     # For debug sake
-    echo "Pod $kata_pod_without_nydus_config file:"
-    cat $kata_pod_without_nydus_config
+    echo "Pod $pod_config file:"
+    cat $pod_config
+
+    # The pod should be failed because the default timeout of CreateContainerRequest is 60s
+    assert_pod_fail "$pod_config"
+    assert_logs_contain "$node" kata "$node_start_time" \
+		'context deadline exceeded'
```

Comment:
Unfortunately this error message is pretty generic and can indicate any error. As we have a test that increases the timeout and it should pass then I think we are fine.

But if we could improve the message (not for this PR)...the problem is that it cannot properly propagate errors within the guest?

Reply:
Yes, I agree. I’ll address it in the upcoming PR and keep you updated.